analyzer/repo-cloner-plugin/src/main/java/eu/fasten/analyzer/repoclonerplugin/RepoClonerPlugin.java

```diff
@@ -0,0 +1,163 @@
+            RepoCloner.baseDir = baseDir;
+        }
+
+        String getRepoPath() {
+            return repoPath;
+        }
+
+        @Override
+        public Optional<List<String>> consumeTopic() {
+            return Optional.of(Collections.singletonList(consumerTopic));
+        }
+
+        @Override
+        public void setTopic(String topicName) {
+            this.consumerTopic = topicName;
+        }
+
+        @Override
+        public Optional<String> produce() {
+            if (artifact == null && group == null) {
+                return Optional.empty();
+            } else {
+                var json = new JSONObject();
+                json.put("artifactId", artifact);
+                json.put("groupId", group);
+                if (version != null && !version.isEmpty()) {
+                    json.put("version", version);
+                }
+                if (repoPath != null) {
+                    json.put("repoPath", repoPath);
+                }
+                return Optional.of(json.toString());
+            }
+        }
+
+        @Override
+        public void consume(String record) {
+            this.pluginError = null;
+            this.artifact = null;
+            this.group = null;
+            this.version = null;
+            this.repoPath = null;
+            var json = new JSONObject(record).getJSONObject("payload");
+            artifact = json.getString("artifactId");
+            group = json.getString("groupId");
+            version = json.optString("version");
+            var repoUrl = json.optString("repoUrl");
+            if (repoUrl != null && !repoUrl.isEmpty()) {
+                try {
+                    var gitCloner = new GitCloner(baseDir);
+                    cloneRepo(repoUrl, gitCloner);
+                } catch (GitAPIException | IOException e) {
+                    logger.error("Error cloning repository for '" + group + ":" + artifact
+                            + ((version == null) ? "" : ":" + version) + "' from " + repoUrl, e);
+                    this.pluginError = e;
+                    return;
+                }
+                if (getPluginError() == null) {
+                    logger.info("Cloned the repo of '" + group + ":" + artifact
+                            + ((version == null) ? "" : ":" + version) + "' from " + repoUrl
+                            + " to " + repoPath);
+                }
+            } else {
+                logger.info("Repository URL not found");
+            }
+        }
+
+        public void cloneRepo(String repoUrl, GitCloner gitCloner)
+                throws GitAPIException, IOException {
+            repoPath = gitCloner.cloneRepo(repoUrl);
+        }
+
+        @Override
+        public String name() {
+            return "Repo Cloner Plugin";
+        }
+
+        @Override
+        public String description() {
+            return "Repo Cloner Plugin. "
+                    + "Consumes GitHub repository URL (if present), "
```

Comment:
It does not consume a GitHub repo URL, does it?

Reply:
It can consume any repository URL which ends with something like `/<repo_owner>/<repo_name>`